MetaMask/metamask-extension
# MV3: contentscript.js - re-activate streams when Service Worker terminates and then resets

## Explanation

When the service worker restarts during a user session, we must reestablish the extension streams.
This PR disconnects, destroys, and recreates the extension streams for the `contentscript.js` when the service worker restarts.

The setup of the streams remains pretty much the same. The logic has been split into separate functions to be reused and to separate concerns. There is additional logic to fire an event when the service worker disconnects to destroy the extension streams and reestablish them.

I had to get rid of the `forwardTrafficBetweenMuxes` functions because we need access to the streams created in these functions. 

## More Information
- Fixes #14845
- Follow-up of #14781

## Screenshots/Screencaps

## Manual Testing Steps

1. Run `yarn start:mv3` 
2. Open test-dapp
3. Check that streams work by using dapp 
4. Terminate the `app-init.js` service worker
   - [chrome://inspect/#service-workers](chrome://inspect/#service-workers)
   - [chrome://serviceworker-internals/](chrome://serviceworker-internals/)
5. Check that dapp continues to work

## Pre-Merge Checklist


### + If there are functional changes:
